cmd/minishift/cmd/start.go

```diff
@@ -221,16 +223,29 @@ func handleProxies() *util.ProxyConfig {
 	return proxyConfig
 }
 
+func getSlice(key string) []string {
+	if viper.IsSet(key) {
+		return viper.GetStringSlice(key)
+	}
+	return nil
+}
+
+func determineInsecureRegistry(key string) []string {
+	s := getSlice(key)
+	return append(s, defaultInsecureRegistry...)
```

Comment:
See the problem is that now we depend on the Docker Daemon doing it, instead of doing it ourself. People might be stumbling over this and wondering the same thing and it is not apparent that de-duping is handled by the Docker Daemon (if we want to rely on this, we probably should add at least a comment). Also, we don't know whether this behaviour is going to change or not. This comes again down to the [Robustness principle](https://en.wikipedia.org/wiki/Robustness_principle) - "Be conservative in what you do, be liberal in what you accept from others".

In this case, conservative means, doing something as simple as de-duping ourselves, making the code explicit and not reliant on things which are not part of any sort of defined API. 

Just explaining where I am coming from here. If you think you want to rely on this de-duping being handled somewhere else, so be it.